gui-libs/display-manager-init/files/display-manager.initd-11

```diff
@@ -0,0 +1,235 @@
+#!/sbin/openrc-run
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License, v2
+
+# This is here to serve as a note to myself, and future developers.
+#
+# Any Display manager (gdm,kdm,xdm) has the following problem:  if
+# it is started before any getty, and no vt is specified, it will
+# usually run on vt2.  When the getty on vt2 then starts, and the
+# DM is already started, the getty will take control of the keyboard,
+# leaving us with a "dead" keyboard.
+#
+# Resolution: add the following line to /etc/inittab
+#
+#  x:a:once:/usr/bin/startDM
+#
+# and have /usr/bin/startDM start the DM in daemon mode if
+# a lock is present (with the info of what DM should be started),
+# else just fall through.
+#
+# How this basically works, is the "a" runlevel is a additional
+# runlevel that you can use to fork processes with init, but the
+# runlevel never gets changed to this runlevel.  Along with the "a"
+# runlevel, the "once" key word means that startDM will only be
+# run when we specify it to run, thus eliminating respawning
+# startDM when "xdm" is not added to the default runlevel, as was
+# done previously.
+#
+# This script then just calls "telinit a", and init will run
+# /usr/bin/startDM after the current runlevel completes (this
+# script should only be added to the actual runlevel the user is
+# using).
+#
+# Martin Schlemmer
+# aka Azarah
+# 04 March 2002
+
+depend() {
+	need localmount display-manager-setup
+
+	# this should start as early as possible
+	# we can't do 'before *' as that breaks it
+	# (#139824) Start after ypbind and autofs for network authentication
+	# (#145219 #180163) Could use lirc mouse as input device
+	# (#70689 comment #92) Start after consolefont to avoid display corruption
+	# (#291269) Start after quota, since some dm need readable home
+	# (#390609) gdm-3 will fail when dbus is not running
+	# (#366753) starting keymaps after X causes problems
+	after bootmisc consolefont modules netmount
+	after readahead-list ypbind autofs openvpn gpm lircmd
+	after quota keymaps
+	before alsasound
+
+	# Start before X
+	use dbus xfs
+}
+
+setup_dm() {
+	local MY_XDM
+	
+	MY_XDM=$(echo "${DISPLAYMANAGER}" | tr '[:upper:]' '[:lower:]')
+
+	# Load our root path from profile.env
+	# Needed for kdm
+	PATH=${PATH}:$(. /etc/profile.env; echo "${ROOTPATH}")
+
+	NAME=
+	case "${MY_XDM}" in
+		kdm|kde)
+			EXE=/usr/bin/kdm
+			PIDFILE=/run/kdm.pid
+			;;
```

Comment:
I wonder why this was not moved into `kdm|kde` condition in the first place back then. I wouldn't drop the condition yet, considering there may still be some users of it, but the `ROOTPATH` loading could be tucked away there.

Reply:
I didn't want to change tooo many things, considering it wasn't humanely possible for me to check with every display manager.

I tested with sddm, gdm and greetd, which seemed a reasonably validating subset.

But changing this sounds good to me :+1:  .

Reply:
done